src/main/java/org/gradle/profiler/ScenarioLoader.java

```diff
@@ -212,80 +217,48 @@ public ScenarioLoader(GradleBuildConfigurationReader gradleBuildConfigurationRea
             String title = scenario.hasPath(TITLE) ? scenario.getString(TITLE) : null;
 
             List<BuildMutator> mutators = BUILD_MUTATOR_CONFIGURATORS.entrySet().stream()
-                .filter(entry -> scenario.hasPath(entry.getKey()))
-                .map(entry -> entry.getValue().configure(scenario, scenarioName, settings, entry.getKey()))
+                .filter(entry -> getBuildConfig(scenario, settings).hasPath(entry.getKey()))
+                .map(entry -> entry.getValue().configure(getBuildConfig(scenario, settings), scenarioName, settings, entry.getKey()))
                 .filter(mutator -> mutator != BuildMutator.NOOP)
                 .collect(Collectors.toList());
 
             int buildCount = getBuildCount(settings, scenario);
             File scenarioBaseDir = new File(settings.getOutputDir(), GradleScenarioDefinition.safeFileName(scenarioName));
 
-            if (scenario.hasPath(BAZEL) && settings.isBazel()) {
```

Comment:
Refactored while I was here to make it more readable.  Extracted this into separate methods instead of a single large method